google-datacatalog-connectors-commons/src/google/datacatalog_connectors/commons/prepare/datacatalog_strings_helper.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/python
+#
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import six
+
+
+class DataCatalogStringsHelper:
+    __UTF8_CHARACTER_ENCODING = 'UTF-8'
+    # String field values are limited to 2000 bytes size when encoded in UTF-8.
+    __STRING_VALUE_UTF8_MAX_LENGTH = 2000
+    __SUFFIX_CHARS_LENGTH = 3
+
+    @classmethod
+    def truncate_string(cls, value, max_length=__STRING_VALUE_UTF8_MAX_LENGTH):
+        """
+        String field values are limited by Data Catalog API at 2000 chars
+        length when encoded in UTF-8. UTF-8 chars may need from 1 to 4 bytes
+        (https://en.wikipedia.org/wiki/UTF-8 for details):
+        - the first 128 characters (US-ASCII) need one byte;
+        - the next 1,920 characters need two bytes to encode, which covers the
+          remainder of almost all Latin-script alphabets, and also Greek,
+          Cyrillic, Coptic, Armenian, Hebrew, Arabic, Syriac, Thaana and N'Ko
+          alphabets, as well as Combining Diacritical Marks;
+        - three bytes are needed for characters in the rest of the Basic
+          Multilingual Plane, which contains virtually all characters in common
+          use, including most Chinese, Japanese and Korean characters;
+        - four bytes are needed for characters in the other planes of Unicode,
+          which include less common CJK characters, various historic scripts,
+          mathematical symbols, and emoji (pictographic symbols).
+
+        Given a value and a string Tag Field, this method assigns the field the
```

Comment:
Please review this docstring. Mentioning a Tag Field here is out of context. Some parts should be kept in the former `_set_string_field` method, others should actually be moved to the refactored code.